RestaurantReservation.Db/Repositories/CustomerRepository.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.IdentityModel.Tokens;
+using RestaurantReservation.Db.Models;
+
+namespace RestaurantReservation.Db.Repositories;
+public class CustomerRepository
+{
+    private RestaurantReservationDbContext _dbContext;
+    public CustomerRepository(RestaurantReservationDbContext dbContext)
+    {
+        _dbContext = dbContext;
+    }
+
+    public async Task<Customer?> AddCustomerAsync(Customer customer)
+    {
+        if (customer == null)
+            return null;
+
+        await _dbContext.Customers.AddAsync(customer);
+        await _dbContext.SaveChangesAsync();
+        return customer;
+    }
+
+    public async Task<bool> RemoveCustomerAsync(int customerId)
```

Comment:
generally deciding whether the operation has to success or fail is service responsibility, so checking for nullable and existence must be in service and repo methods don't need to return Boolean just do their jobs ( delete, update, and so on).